### ISSUE TITLE
Block Directory: Missing styles when installing blocks from the inserter

### ISSUE THREAD

**Issue by safar-baig** [WordPress/gutenberg#63762]
### Description

Hi, I have a block plugin called “Safelayout Elegant Icons”. The plugin is hosted in the WordPress repository and provide two blocks, icon block and icon box block.
When I install the plugin from the plugin section of the WordPress dashboard, everything is fine, **the problem occurs when I install the plugin within the block editor itself.**

After installing the plugin, the CSS style does not work properly and I have to reload the page to fix the problem. And this makes the user think that the plugin always acts like this and uninstalls the plugin.

Because the block editor places blocks inside an iframe tag( **<iframe name=”editor-canvas”** ), plugin CSS files loaded while the block is installing inside the block editor itself, do not affect the blocks inside the iframe tag.

### Step-by-step reproduction instructions

1- Create a new post.
2- Click on the “Toggle block inserter” button at the top left side of the page.
3- Type “Icon (Safelayout)” in the search box.
4- After displaying the plugin at the bottom of the search list, Click on it to install it.

### Screenshots, screen recording, code snippet

![1](https://github.com/user-attachments/assets/11fbae80-5fee-4148-ac35-2a3190811074)
![2](https://github.com/user-attachments/assets/d3345aff-4c3a-4889-b728-3109b222c2d3)

### Environment info

WordPress Verson: 6.6

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by ndiego** [comment]
Yes, I can confirm. This issue has existed for some time. I meant to add an issue about this a while ago 🤦‍♂️

**Comment by gziolo** [comment]
That sounds like a legitimate bug. As far as I know, when installing a block from Block Directory scripts and styles get installed in the main frame with:

https://github.com/WordPress/gutenberg/blob/159d01a01b61ee76159ab67bc94d0d7cb3d90b35/packages/block-directory/src/store/load-assets.js#L16

@ellatrix, can you leave some pointers how the iframe could get updated to include relevant CSS code?

**Comment by Lovor01** [comment]
Any progress on this?
It also happens with my block plugin:
https://wordpress.org/plugins/makeiteasy-popup/
